src/useTimer.ts

```diff
@@ -19,92 +19,58 @@ export interface IValues {
   time: number;
 }
 
-const initialConfig: IConfig = {
-  endTime: null,
-  initialTime: 0,
-  interval: 1000,
-  step: 1,
-  timerType: 'INCREMENTAL',
-};
-
-export const useTimer = (config?: Partial<IConfig>): IValues => {
-  const { endTime, initialTime, interval, onTimeOver, step, timerType } = {
-    ...initialConfig,
-    ...config,
-  };
-  const intervalRef = useRef<NodeJS.Timeout | null>(null);
-  const pausedTimeRef = useRef<number | null>(null);
-  const [shouldResetTime, setShouldResetTime] = useState(false);
+export const useTimer = ({
+  initialTime = 0,
+  interval = 1000,
+  step = 1,
+  timerType = 'INCREMENTAL',
+  endTime,
+  onTimeOver,
+}: Partial<IConfig> = {}): IValues => {
   const [time, setTime] = useState(initialTime);
   const [isRunning, setIsRunning] = useState(false);
+  const [isCountFinished, setIsCountFinished] = useState(false);
```

Comment:
I would prefer to name it **isTimeOver**, to be coherent with the **onTimeOver** function.

```suggestion
  const [isTimeOver, setIsTimeOver] = useState(false);
```